plugins/docker/docker_cpu

```diff
@@ -100,11 +103,14 @@ if (defined $ARGV[0] and $ARGV[0] eq "config")
 
    foreach(@result)
    {
-      print "$$_{'name'}.label $$_{'name'}\n";
-      print "$$_{'name'}.draw LINE2\n";
-      print "$$_{'name'}.min 0\n";
-      print "$$_{'name'}.type DERIVE\n";
-      print "$$_{'name'}.cdef $$_{'name'},$nanoSecondsInSecond,/\n";
+      if ($$_{'name'})
```

Comment:
How could it happen, that this variable is empty?

Reply:
that was a configuration / system problem (munin-node had no access to the necessary paths (i did not map all the docker paths into my docker-container running munin-node)), i guess it can happen on other systems as well if the process returns an error.

Reply:
Hm. I would feel better, if this problem could be detected when it happens, instead of cleaning in up in the output.
But if you have no idea, how this could be done in the plugin, then I would appreciate a comment above the `if`, which describes the problem, that it tries to solve.

Thank you!

Reply:
I'll add the comment (in the code).

For me with munin-node plugins i always try to separate real errors from perl warnings (i find it very hard to diagnose the root cause if an error happens and there is a bunch of perl warnings for missing vars as well). so this cleans up the perl warnings when an error happens. the error is still shown but the warnings are gone. so in case of an error you now only get the error and this makes it easier for me to focus on the error and not on the warnings.

Reply:
ok the warnings are even more anoying, without any error or configuration issue, this happens if there is no docker containers running (without the if):
```
bigbear3001@wt-io-it-bigbear3001:/etc/munin/plugins$ sudo munin-run docker_cpu config
graph_title Docker container CPU usage
graph_args --base 1000 -r --lower-limit 0 --upper-limit 100
graph_vlabel %
graph_scale no
graph_period second
graph_category virtualization
graph_info This graph shows docker container CPU usage.
Use of uninitialized value in concatenation (.) or string at /etc/munin/plugins/docker_cpu line 106.
Use of uninitialized value in concatenation (.) or string at /etc/munin/plugins/docker_cpu line 106.
.label 
Use of uninitialized value in concatenation (.) or string at /etc/munin/plugins/docker_cpu line 107.
.draw LINE2
Use of uninitialized value in concatenation (.) or string at /etc/munin/plugins/docker_cpu line 108.
.min 0
Use of uninitialized value in concatenation (.) or string at /etc/munin/plugins/docker_cpu line 109.
.type DERIVE
Use of uninitialized value in concatenation (.) or string at /etc/munin/plugins/docker_cpu line 110.
Use of uninitialized value in concatenation (.) or string at /etc/munin/plugins/docker_cpu line 110.
.cdef ,1000000000,/
```

Reply:
found a better fix:
```
my $graphlimit = $result[0]{'ncpu'}
```
somehow initialized $result / @ result
and as there was a loop right after that, replacing it with an initial value of 1 should have the same result
```
   my $graphlimit = 1;
   foreach(@result){
      if ($$_{'ncpu'} || 1 > $graphlimit){
         $graphlimit = $$_{'ncpu'};
      }
   }
```

sorry that i didn't catch that right away (perl isn't one of my common languages) and thanks for sticking with the problem. i hope that this solution is more appropriate.

Reply:
I guess, I found the culprit:
```
my $graphlimit = $result[0]{'ncpu'};
```

The above line accesses a non-existing array item and turns it into a hash. Perl can be astonishingly weird :)

If you replace the line above with `my $graphlimit = 1;`, then your sanity check can be removed.